lib/linguist/heuristics.yml

```diff
@@ -195,6 +195,10 @@ disambiguations:
     #  : dependency
     # path/file.ext1 : some/path/../file.ext2
     pattern: '([\/\\].*:\s+.*\s\\$|: \\$|^[ %]:|^[\w\s\/\\.]+\w+\.\w+\s*:\s+[\w\s\/\\.]+\w+\.\w+)'
+- extensions: ['.def']
+  rules:
+  - language: Modula-2
+    pattern: '^\s*(?i:DEFINITION\s+MODULE|END)\s+[\w\.]+\s*;'
```

Comment:
This heuristic will never be used as this is the only user of this extension within linguist which in itself is potentially a problem as `.def` is an incredibly generic extension used by many things.

If you're 100% certain this regex will only match Modula-2 files, then adding this extension to the `generic.yml` file will ensure this regex takes effect and is used. If there's a risk this misidentifies other languages, it's best to leave off this extension and use an override when needed.